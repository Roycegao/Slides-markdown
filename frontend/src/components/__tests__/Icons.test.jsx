import { describe, it, expect } from 'vitest';
import { render } from '@testing-library/react';
import {
  AddIcon,
  RemoveIcon,
  FullscreenIcon,
  FullscreenExitIcon,
  PrevIcon,
  NextIcon,
  UpIcon,
  DownIcon,
  BackIcon,
  CloseIcon
} from '../Icons';

describe('Icons', () => {
  it('renders AddIcon', () => {
    const { container } = render(<AddIcon />);
    expect(container.firstChild).toMatchSnapshot();
  });
  it('renders RemoveIcon', () => {
    const { container } = render(<RemoveIcon />);
    expect(container.firstChild).toMatchSnapshot();
  });
  it('renders FullscreenIcon', () => {
    const { container } = render(<FullscreenIcon />);
    expect(container.firstChild).toMatchSnapshot();
  });
  it('renders FullscreenExitIcon', () => {
    const { container } = render(<FullscreenExitIcon />);
    expect(container.firstChild).toMatchSnapshot();
  });
  it('renders PrevIcon', () => {
    const { container } = render(<PrevIcon />);
    expect(container.firstChild).toMatchSnapshot();
  });
  it('renders NextIcon', () => {
    const { container } = render(<NextIcon />);
    expect(container.firstChild).toMatchSnapshot();
  });
  it('renders UpIcon', () => {
    const { container } = render(<UpIcon />);
    expect(container.firstChild).toMatchSnapshot();
  });
  it('renders DownIcon', () => {
    const { container } = render(<DownIcon />);
    expect(container.firstChild).toMatchSnapshot();
  });
  it('renders BackIcon', () => {
    const { container } = render(<BackIcon />);
    expect(container.firstChild).toMatchSnapshot();
  });
  it('renders CloseIcon', () => {
    const { container } = render(<CloseIcon />);
    expect(container.firstChild).toMatchSnapshot();
  });
}); 