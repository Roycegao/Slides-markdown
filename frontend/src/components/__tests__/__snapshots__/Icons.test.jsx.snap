// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Icons > renders AddIcon 1`] = `
<svg
  class=""
  fill="currentColor"
  viewBox="0 0 24 24"
>
  <path
    d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
  />
</svg>
`;

exports[`Icons > renders BackIcon 1`] = `
<svg
  class=""
  fill="currentColor"
  viewBox="0 0 24 24"
>
  <path
    d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
  />
</svg>
`;

exports[`Icons > renders CloseIcon 1`] = `
<svg
  class=""
  fill="currentColor"
  viewBox="0 0 24 24"
>
  <path
    d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
  />
</svg>
`;

exports[`Icons > renders DownIcon 1`] = `
<svg
  class=""
  fill="currentColor"
  viewBox="0 0 24 24"
>
  <path
    d="M7.41 8.59L12 13.17l4.59-4.58L18 10l-6 6-6-6z"
  />
</svg>
`;

exports[`Icons > renders FullscreenExitIcon 1`] = `
<svg
  class=""
  fill="currentColor"
  viewBox="0 0 24 24"
>
  <path
    d="M5 16h3v3h2v-5H5v2zm3-8H5v2h5V5H8v3zm6 11h2v-3h3v-2h-5v5zm2-11V5h-2v5h5V8h-3z"
  />
</svg>
`;

exports[`Icons > renders FullscreenIcon 1`] = `
<svg
  class=""
  fill="currentColor"
  viewBox="0 0 24 24"
>
  <path
    d="M7 14H5v5h5v-2H7v-3zm-2-4h2V7h3V5H5v5zm12 7h-3v2h5v-5h-2v3zM14 5v2h3v3h2V5h-5z"
  />
</svg>
`;

exports[`Icons > renders NextIcon 1`] = `
<svg
  class=""
  fill="currentColor"
  viewBox="0 0 24 24"
>
  <path
    d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
  />
</svg>
`;

exports[`Icons > renders PrevIcon 1`] = `
<svg
  class=""
  fill="currentColor"
  viewBox="0 0 24 24"
>
  <path
    d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
  />
</svg>
`;

exports[`Icons > renders RemoveIcon 1`] = `
<svg
  class=""
  fill="currentColor"
  viewBox="0 0 24 24"
>
  <path
    d="M19 13H5v-2h14v2z"
  />
</svg>
`;

exports[`Icons > renders UpIcon 1`] = `
<svg
  class=""
  fill="currentColor"
  viewBox="0 0 24 24"
>
  <path
    d="M7.41 15.41L12 10.83l4.59 4.58L18 14l-6-6-6 6z"
  />
</svg>
`;
